docs/HyperIndex/Guides/event-handlers.mdx

```diff
@@ -65,62 +65,93 @@ Handlers.<ContractName>.<EventName>.handler(async ({event, context}) => {
+
+```typescript
+type TransactionFields = {
+  hash: string;
+  transactionIndex: number;
+};
+
+type AdditionalBlockFields = {
+  parentHash: string;
+};
+
+type BlockFields = {
+  // number, timestamp, and hash are always available by default
+  number: number;
+  timestamp: number;
+  hash: string;
+} & AdditionalBlockFields;
+```
+
+It is recommended to add `# yaml-language-server: $schema=../../codegenerator/cli/npm/envio/evm.schema.json` to the top of your `config.yaml` to help with editor autocompletion.
 
 ### Context
 
-In your handler functions you can access entities in the database via the asyncronous 'get' function as shown below:
+In your handler functions, you can access entities in the database via the asynchronous `get` function as shown below:
 
 ```javascript
-await context.<EntityName>.get(<id>)
+await context.<ENTITY_NAME>.get(<ID>);
 ```
 
 The context also contains some logging functions that include the context automatically. You can also use `console.log`.
 
-> Dev note: 📢 For indexers built using `ReScript`, use a lower case for first letter of entityName when accessing it via context (i.e. `context.user` instead of `context.User`).
+> Dev note: 📢 For indexers built using `ReScript`, use a lower case for the first letter of `entityName` when accessing it via context (i.e., `context.user` instead of `context.User`).
 
-In addition to the 'get' which is a read only action, two write actions are provided to you in the context:
+In addition to the `get` function, which is a read-only action, two write actions are provided in the context:
 
-- set
-- deleteUnsafe
+- `set`
+- `deleteUnsafe`
 
-Used as follows
+Used as follows:
 
 ```javascript
-context.<entityName>.set(<entityObject>)
+context.<ENTITY_NAME>.set(<ENTITY_OBJECT>);
 ```
 
 and
 
 ```javascript
-context.<EntityName>.deleteUnsafe(<entityObject>.id)
+context.<ENTITY_NAME>.deleteUnsafe(<ENTITY_OBJECT>.id);
 ```
 
-The `set` method is used to either create an entity or to update an existing entity with the values defined in the `entityObject`.
-The `deleteUnsafe` method is an experimental feature that allows you to remove an entity with a particular ID from the database and indexer memory. This is an **unsafe** function,
-since there may be unexpected repercussions when a deleted entity is linked to another entity. It is your responsibility to delete/fix these references manually since a 'safe' delete function hasn't been created yet.
+The `set` method is used to either create an entity or to update an existing entity with the values defined in the `entityObject`. The `deleteUnsafe` method is an experimental feature that allows you to remove an entity with a particular ID from the database and indexer memory. This is an **unsafe** function since there may be unexpected repercussions when a deleted entity is linked to another entity. It is your responsibility to delete/fix these references manually since a 'safe' delete function hasn't been created yet.
```

Comment:
I think we can cut the sentence here 😅
```suggestion
The `set` method is used to either create an entity or to update an existing entity with the values defined in the `entityObject`. The `deleteUnsafe` method is an experimental feature that allows you to remove an entity with a particular ID from the database and indexer memory. This is an **unsafe** function since there may be unexpected repercussions when a deleted entity is linked to another entity. It is your responsibility to delete/fix these references manually.
```